api/src/paths/alert/index.ts

```diff
@@ -0,0 +1,244 @@
+  tags: ['alerts'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      in: 'query',
+      name: 'types',
+      required: false,
+      schema: {
+        type: 'array',
+        items: {
+          type: 'string',
+          nullable: true
+        }
+      }
+    },
+    {
+      in: 'query',
+      name: 'expiresBefore',
+      required: false,
+      schema: {
+        type: 'string',
+        nullable: true
+      }
+    },
+    {
+      in: 'query',
+      name: 'expiresAfter',
+      required: false,
+      schema: {
+        type: 'string',
+        nullable: true
+      }
+    }
+  ],
+  responses: {
+    200: {
+      description: 'System alert response object',
+      content: {
+        'application/json': {
+          schema: {
+            type: 'object',
+            description: 'Response object containing system alerts',
+            additionalProperties: false,
+            required: ['alerts'],
+            properties: { alerts: { type: 'array', description: 'Array of system alerts', items: systemAlertSchema } }
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/403'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+/**
+ * Get system alerts created by system administrators describing important information, deadlines, etc.
+ *
+ * @returns {RequestHandler}
+ */
+export function getAlerts(): RequestHandler {
+  return async (req, res) => {
+    defaultLog.debug({ label: 'getAlerts' });
+
+    const connection = getDBConnection(req.keycloak_token);
+
+    try {
+      await connection.open();
+
+      const filterObject = parseQueryParams(req);
+
+      const alertService = new AlertService(connection);
+
+      const alerts = await alertService.getAlerts(filterObject);
+
+      await connection.commit();
+
+      return res.status(200).json({ alerts: alerts });
+    } catch (error) {
+      defaultLog.error({ label: 'getAlerts', message: 'error', error });
+      await connection.rollback();
+      throw error;
+    } finally {
+      connection.release();
+    }
+  };
+}
+
+/**
+ * Parse the query parameters from the request into the expected format.
+ *
+ * @param {Request<unknown, unknown, unknown, IAlertFilterObject>} req
+ * @return {*}  {IAlertFilterObject}
+ */
+function parseQueryParams(req: Request<unknown, unknown, unknown, IAlertFilterObject>): IAlertFilterObject {
+  return {
+    expiresBefore: req.query.expiresBefore ?? undefined,
+    expiresAfter: req.query.expiresAfter ?? undefined,
+    types: req.query.types ?? []
+  };
+}
+
+export const POST: Operation = [
+  authorizeRequestHandler(() => {
+    return {
+      and: [
+        {
+          discriminator: 'SystemUser',
```

Comment:
This should be `'SystemRole'`

There are a few other endpoints I notice with the same issue.

Searching `discriminator: 'SystemUser',` seems to find the other 3 I can see.